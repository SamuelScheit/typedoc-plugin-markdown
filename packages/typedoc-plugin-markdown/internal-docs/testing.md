---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
