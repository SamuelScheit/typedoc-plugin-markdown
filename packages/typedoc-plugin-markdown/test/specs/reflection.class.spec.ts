import { expectFileToEqual } from '@devtools/testing';

describe(`Class Reflection`, () => {
  test(`should compile abstract class`, () => {
    expectFileToEqual('reflections', 'members', 'classes/AbstractClass.md');
  });

  test(`should compile derived class`, () => {
    expectFileToEqual('reflections', 'members', 'classes/DerivedClassA.md');
  });

  test(`should compile class with constructor overloads`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'classes/ClassWithConstructorOverloads.md',
    );
  });

  test(`should compile class with type parameters`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'classes/ClassWithTypeParameters.md',
    );
  });

  test(`should compile class with accessors`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'classes/ClassWithAccessors.md',
    );
  });

  test(`should compile class with modifiers`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'classes/ClassWithModifiers.md',
    );
  });

  test(`should compile class with flags`, () => {
    expectFileToEqual('reflections', 'members', 'classes/ClassWithFlags.md');
  });

  test(`should compile disposable class`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'classes/DisposableClass.md',
      1,
    );
  });
});
