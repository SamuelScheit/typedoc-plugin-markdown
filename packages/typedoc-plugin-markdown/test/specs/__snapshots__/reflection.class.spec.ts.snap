// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class Reflection should compile abstract class: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: \`abstract\` AbstractClass

## Extended by

- [\`DerivedClassA\`](DerivedClassA.md)
- [\`DerivedClassB\`](DerivedClassB.md)

## Constructors

### new AbstractClass()

> **new AbstractClass**(): [\`AbstractClass\`](AbstractClass.md)

#### Returns

[\`AbstractClass\`](AbstractClass.md)

## Properties

### abstractProp

> **\`abstract\`** **abstractProp**: \`string\`

Comments for abstractProp

#### Source

[classes.ts:9](http://source-url)

## Methods

### abstractMethod()

> **\`abstract\`** **abstractMethod**(): \`string\`

Comments for abstractMethod

#### Returns

\`string\`

#### Source

[classes.ts:14](http://source-url)
"
`;

exports[`Class Reflection should compile abstract class: (Output File Strategy "members") (Option Group "2") 1`] = `
"# \`abstract\` AbstractClass

## Extended by

- [\`DerivedClassA\`](DerivedClassA.md)
- [\`DerivedClassB\`](DerivedClassB.md)

## Constructors

### new AbstractClass()

\`\`\`ts
new AbstractClass(): AbstractClass
\`\`\`

#### Returns

[\`AbstractClass\`](AbstractClass.md)

## Properties

| Property | Modifier | Type | Description |
| :------ | :------ | :------ | :------ |
| \`abstractProp\` | \`abstract\` | \`string\` | Comments for abstractProp |

## Methods

### abstractMethod()

\`\`\`ts
abstract abstractMethod(): string
\`\`\`

Comments for abstractMethod

#### Returns

\`string\`

#### Source

[classes.ts:14](http://source-url)
"
`;

exports[`Class Reflection should compile class with accessors: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: ClassWithAccessors

Comments for ClassWithAccessors

## Constructors

### new ClassWithAccessors()

> **new ClassWithAccessors**(): [\`ClassWithAccessors\`](ClassWithAccessors.md)

#### Returns

[\`ClassWithAccessors\`](ClassWithAccessors.md)

## Properties

### privateProp

> **\`private\`** **privateProp**: \`string\`

#### Source

[classes.ts:72](http://source-url)

## Accessors

### accessor

> **\`get\`** **accessor**(): \`string\`

Comments for getter

> **\`set\`** **accessor**(\`value\`): \`void\`

Comments for setter

#### Parameters

• **value**: \`string\`

Param comments

#### Returns

\`string\`

#### Source

[classes.ts:76](http://source-url)

***

### noGetter

> **\`set\`** **noGetter**(\`x\`): \`void\`

#### Parameters

• **x**: \`string\`

#### Source

[classes.ts:92](http://source-url)

***

### noSetter

> **\`get\`** **noSetter**(): \`string\`

#### Returns

\`string\`

#### Source

[classes.ts:88](http://source-url)
"
`;

exports[`Class Reflection should compile class with accessors: (Output File Strategy "members") (Option Group "2") 1`] = `
"# ClassWithAccessors

Comments for ClassWithAccessors

## Constructors

### new ClassWithAccessors()

\`\`\`ts
new ClassWithAccessors(): ClassWithAccessors
\`\`\`

#### Returns

[\`ClassWithAccessors\`](ClassWithAccessors.md)

## Properties

| Property | Modifier | Type |
| :------ | :------ | :------ |
| \`privateProp\` | \`private\` | \`string\` |

## Accessors

### accessor

\`\`\`ts
get accessor(): string
\`\`\`

Comments for getter

\`\`\`ts
set accessor(value): void
\`\`\`

Comments for setter

#### Parameters

| Parameter | Type | Description |
| :------ | :------ | :------ |
| \`value\` | \`string\` | Param comments |

#### Returns

\`string\`

#### Source

[classes.ts:76](http://source-url)

***

### noGetter

\`\`\`ts
set noGetter(x): void
\`\`\`

#### Parameters

| Parameter | Type |
| :------ | :------ |
| \`x\` | \`string\` |

#### Source

[classes.ts:92](http://source-url)

***

### noSetter

\`\`\`ts
get noSetter(): string
\`\`\`

#### Returns

\`string\`

#### Source

[classes.ts:88](http://source-url)
"
`;

exports[`Class Reflection should compile class with constructor overloads: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: ClassWithConstructorOverloads

Comments for ClassWithConstructorOverloads

## Constructors

### new ClassWithConstructorOverloads(x, y)

> **new ClassWithConstructorOverloads**(\`x\`, \`y\`): [\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Parameters

• **x**: \`number\`

Comments for x number

• **y**: \`string\`

#### Returns

[\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Source

[classes.ts:50](http://source-url)

### new ClassWithConstructorOverloads(x)

> **new ClassWithConstructorOverloads**(\`x\`): [\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Parameters

• **x**: \`string\`

Comments for x string

#### Returns

[\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Source

[classes.ts:55](http://source-url)

### new ClassWithConstructorOverloads(s)

> **new ClassWithConstructorOverloads**(\`s\`): [\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Parameters

• **s**: \`string\`

#### Returns

[\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Source

[classes.ts:56](http://source-url)
"
`;

exports[`Class Reflection should compile class with constructor overloads: (Output File Strategy "members") (Option Group "2") 1`] = `
"# ClassWithConstructorOverloads

Comments for ClassWithConstructorOverloads

## Constructors

### new ClassWithConstructorOverloads(x, y)

\`\`\`ts
new ClassWithConstructorOverloads(x, y): ClassWithConstructorOverloads
\`\`\`

#### Parameters

| Parameter | Type | Description |
| :------ | :------ | :------ |
| \`x\` | \`number\` | Comments for x number |
| \`y\` | \`string\` |  |

#### Returns

[\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Source

[classes.ts:50](http://source-url)

### new ClassWithConstructorOverloads(x)

\`\`\`ts
new ClassWithConstructorOverloads(x): ClassWithConstructorOverloads
\`\`\`

#### Parameters

| Parameter | Type | Description |
| :------ | :------ | :------ |
| \`x\` | \`string\` | Comments for x string |

#### Returns

[\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Source

[classes.ts:55](http://source-url)

### new ClassWithConstructorOverloads(s)

\`\`\`ts
new ClassWithConstructorOverloads(s): ClassWithConstructorOverloads
\`\`\`

#### Parameters

| Parameter | Type |
| :------ | :------ |
| \`s\` | \`string\` |

#### Returns

[\`ClassWithConstructorOverloads\`](ClassWithConstructorOverloads.md)

#### Source

[classes.ts:56](http://source-url)
"
`;

exports[`Class Reflection should compile class with flags: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: ClassWithFlags

Comment for ClassWithFlags

## Constructors

### new ClassWithFlags()

> **new ClassWithFlags**(): [\`ClassWithFlags\`](ClassWithFlags.md)

#### Returns

[\`ClassWithFlags\`](ClassWithFlags.md)

## Properties

### expermintalProp

\`Experimental\`

> **expermintalProp**: \`string\`

#### Source

[classes.ts:157](http://source-url)

***

### internalProp

\`Internal\`

> **\`private\`** **internalProp**: \`string\`

#### Source

[classes.ts:159](http://source-url)

## Methods

### expermintalMethod()

\`Experimental\`

> **expermintalMethod**(): \`void\`

#### Returns

\`void\`

#### Source

[classes.ts:161](http://source-url)

***

### internalMethod()

\`Internal\`

> **\`protected\`** **internalMethod**(): \`void\`

#### Returns

\`void\`

#### Source

[classes.ts:165](http://source-url)
"
`;

exports[`Class Reflection should compile class with flags: (Output File Strategy "members") (Option Group "2") 1`] = `
"# ClassWithFlags

Comment for ClassWithFlags

## Constructors

### new ClassWithFlags()

\`\`\`ts
new ClassWithFlags(): ClassWithFlags
\`\`\`

#### Returns

[\`ClassWithFlags\`](ClassWithFlags.md)

## Properties

| Property | Modifier | Flags | Type |
| :------ | :------ | :------ | :------ |
| \`expermintalProp\` | \`public\` | \`Experimental\` | \`string\` |
| \`internalProp\` | \`private\` | \`Internal\` | \`string\` |

## Methods

### expermintalMethod()

\`Experimental\`

\`\`\`ts
expermintalMethod(): void
\`\`\`

#### Returns

\`void\`

#### Source

[classes.ts:161](http://source-url)

***

### internalMethod()

\`Internal\`

\`\`\`ts
protected internalMethod(): void
\`\`\`

#### Returns

\`void\`

#### Source

[classes.ts:165](http://source-url)
"
`;

exports[`Class Reflection should compile class with modifiers: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: \`abstract\` ClassWithModifiers

Comments for ClassWithModifiers

## Constructors

### new ClassWithModifiers()

> **new ClassWithModifiers**(): [\`ClassWithModifiers\`](ClassWithModifiers.md)

#### Returns

[\`ClassWithModifiers\`](ClassWithModifiers.md)

## Properties

### privateProp

> **\`private\`** **privateProp**: \`string\`

Comments for privateProp

#### Source

[classes.ts:111](http://source-url)

***

### protectedProp

> **\`protected\`** **protectedProp**: \`string\`

Comments for protectedProp

#### Source

[classes.ts:121](http://source-url)

***

### publicPropWithDefault

> **publicPropWithDefault**: \`string\` = \`'propWithDefault'\`

Comments for propWithDefault

#### Source

[classes.ts:126](http://source-url)

***

### readonlyProp

> **\`readonly\`** **readonlyProp**: \`string\`

Comments for abstractProperty

#### Source

[classes.ts:116](http://source-url)

***

### staticProp

> **\`static\`** **staticProp**: \`string\`

Comments for staticProp

#### Source

[classes.ts:106](http://source-url)

## Methods

### privateMethod()

> **\`private\`** **privateMethod**(): \`void\`

Comment for privateMethod

#### Returns

\`void\`

#### Source

[classes.ts:138](http://source-url)

***

### publicMethod()

> **publicMethod**(): \`void\`

Comment for publicMethod

#### Returns

\`void\`

#### Source

[classes.ts:131](http://source-url)

***

### staticMethod()

> **\`static\`** **staticMethod**(): \`void\`

Comment for staticMethod

#### Returns

\`void\`

#### Source

[classes.ts:143](http://source-url)
"
`;

exports[`Class Reflection should compile class with modifiers: (Output File Strategy "members") (Option Group "2") 1`] = `
"# \`abstract\` ClassWithModifiers

Comments for ClassWithModifiers

## Constructors

### new ClassWithModifiers()

\`\`\`ts
new ClassWithModifiers(): ClassWithModifiers
\`\`\`

#### Returns

[\`ClassWithModifiers\`](ClassWithModifiers.md)

## Properties

| Property | Modifier | Type | Description |
| :------ | :------ | :------ | :------ |
| \`privateProp\` | \`private\` | \`string\` | Comments for privateProp |
| \`protectedProp\` | \`protected\` | \`string\` | Comments for protectedProp |
| \`publicPropWithDefault\` | \`public\` | \`string\` | Comments for propWithDefault |
| \`readonlyProp\` | \`readonly\` | \`string\` | Comments for abstractProperty |
| \`staticProp\` | \`static\` | \`string\` | Comments for staticProp |

## Methods

### privateMethod()

\`\`\`ts
private privateMethod(): void
\`\`\`

Comment for privateMethod

#### Returns

\`void\`

#### Source

[classes.ts:138](http://source-url)

***

### publicMethod()

\`\`\`ts
publicMethod(): void
\`\`\`

Comment for publicMethod

#### Returns

\`void\`

#### Source

[classes.ts:131](http://source-url)

***

### staticMethod()

\`\`\`ts
static staticMethod(): void
\`\`\`

Comment for staticMethod

#### Returns

\`void\`

#### Source

[classes.ts:143](http://source-url)
"
`;

exports[`Class Reflection should compile class with type parameters: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: ClassWithTypeParameters\\<A, B, C\\>

Comments for ClassWithTypeParameters

## Type parameters

• **A**

Comments for param A

• **B** extends \`string\`

Comments for param B

• **C** = \`boolean\`

## Constructors

### new ClassWithTypeParameters()

> **new ClassWithTypeParameters**\\<\`A\`, \`B\`, \`C\`\\>(): [\`ClassWithTypeParameters\`](ClassWithTypeParameters.md)\\<\`A\`, \`B\`, \`C\`\\>

#### Returns

[\`ClassWithTypeParameters\`](ClassWithTypeParameters.md)\\<\`A\`, \`B\`, \`C\`\\>
"
`;

exports[`Class Reflection should compile class with type parameters: (Output File Strategy "members") (Option Group "2") 1`] = `
"# ClassWithTypeParameters\\<A, B, C\\>

Comments for ClassWithTypeParameters

## Type parameters

| Type parameter | Value | Description |
| :------ | :------ | :------ |
| \`A\` | - | Comments for param A |
| \`B\` extends \`string\` | - | Comments for param B |
| \`C\` | \`boolean\` | - |

## Constructors

### new ClassWithTypeParameters()

\`\`\`ts
new ClassWithTypeParameters<A, B, C>(): ClassWithTypeParameters<A, B, C>
\`\`\`

#### Returns

[\`ClassWithTypeParameters\`](ClassWithTypeParameters.md)\\<\`A\`, \`B\`, \`C\`\\>
"
`;

exports[`Class Reflection should compile derived class: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: DerivedClassA

Comments for DerivedClassA

## Extends

- [\`AbstractClass\`](AbstractClass.md)

## Constructors

### new DerivedClassA()

> **new DerivedClassA**(): [\`DerivedClassA\`](DerivedClassA.md)

#### Returns

[\`DerivedClassA\`](DerivedClassA.md)

#### Inherited from

[\`AbstractClass\`](AbstractClass.md).[\`constructor\`](AbstractClass.md#constructors)

## Properties

### abstractProp

> **abstractProp**: \`string\`

Comments for abstractProp

#### Overrides

[\`AbstractClass\`](AbstractClass.md).[\`abstractProp\`](AbstractClass.md#abstractprop)

#### Source

[classes.ts:21](http://source-url)

***

### derivedProp

> **derivedProp**: \`string\`

#### Source

[classes.ts:22](http://source-url)

## Methods

### abstractMethod()

> **abstractMethod**(): \`string\`

Comments for abstractMethod

#### Returns

\`string\`

#### Overrides

[\`AbstractClass\`](AbstractClass.md).[\`abstractMethod\`](AbstractClass.md#abstractmethod)

#### Source

[classes.ts:23](http://source-url)

***

### derivedMethod()

> **derivedMethod**(): \`string\`

#### Returns

\`string\`

#### Source

[classes.ts:26](http://source-url)
"
`;

exports[`Class Reflection should compile derived class: (Output File Strategy "members") (Option Group "2") 1`] = `
"# DerivedClassA

Comments for DerivedClassA

## Extends

- [\`AbstractClass\`](AbstractClass.md)

## Constructors

### new DerivedClassA()

\`\`\`ts
new DerivedClassA(): DerivedClassA
\`\`\`

#### Returns

[\`DerivedClassA\`](DerivedClassA.md)

#### Inherited from

[\`AbstractClass\`](AbstractClass.md).[\`constructor\`](AbstractClass.md#constructors)

## Properties

| Property | Type | Description | Overrides |
| :------ | :------ | :------ | :------ |
| \`abstractProp\` | \`string\` | Comments for abstractProp | [\`AbstractClass\`](AbstractClass.md).\`abstractProp\` |
| \`derivedProp\` | \`string\` | - | - |

## Methods

### abstractMethod()

\`\`\`ts
abstractMethod(): string
\`\`\`

Comments for abstractMethod

#### Returns

\`string\`

#### Overrides

[\`AbstractClass\`](AbstractClass.md).[\`abstractMethod\`](AbstractClass.md#abstractmethod)

#### Source

[classes.ts:23](http://source-url)

***

### derivedMethod()

\`\`\`ts
derivedMethod(): string
\`\`\`

#### Returns

\`string\`

#### Source

[classes.ts:26](http://source-url)
"
`;

exports[`Class Reflection should compile disposable class: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Class: DisposableClass

## Implements

- \`Disposable\`

## Constructors

### new DisposableClass()

> **new DisposableClass**(): [\`DisposableClass\`](DisposableClass.md)

#### Returns

[\`DisposableClass\`](DisposableClass.md)

## Methods

### \`[dispose]\`()

> **[dispose]**(): \`void\`

#### Returns

\`void\`

#### Implementation of

\`Disposable.[dispose]\`

#### Source

[classes.ts:147](http://source-url)
"
`;
